common/app/uiComponent/core/JavascriptEngine.scala

```diff
@@ -0,0 +1,25 @@
+package uiComponent.core
+
+import java.io.Reader
+import javax.script._
+
+import jdk.nashorn.api.scripting.JSObject
+
+import scala.util.Try
+
+class JavascriptEngine() {
+  import JavascriptEngine._
+
+  private lazy val bindings = shared.createBindings()
```

Comment:
I read the code as `JavascriptEngine` is a singleton, shared by each script evaluation. So if two evaluations or `call` invocations occur on different threads, will they use the same instance of bindings? I can see now the `SimpleScriptContext`, which means each context has it's own bindings. Wasn't sure if that was the case in the previous version.